src/python/pants/backend/python/rules/coverage_test.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from textwrap import dedent
+from typing import List, Optional
+
+import pytest
+
+from pants.backend.python.rules.coverage import CoverageSubsystem, create_coverage_config
+from pants.engine.fs import CreateDigest, Digest, DigestContents, FileContent, PathGlobs, Snapshot
+from pants.testutil.engine.util import MockGet, create_subsystem, run_rule
+from pants.testutil.test_base import TestBase
+
+
+class TestCoverageConfig(TestBase):
+    def run_create_coverage_config_rule(self, coverage_config: Optional[str]) -> str:
+        coverage = create_subsystem(
+            CoverageSubsystem, config="some_file" if coverage_config else None
+        )
+        resolved_config: List[str] = []
+
+        def fake_handle_config(fcs):
+            assert len(fcs) == 1
+            assert fcs[0].path == ".coveragerc"
+            assert fcs[0].is_executable is False
+            resolved_config.append(fcs[0].content.decode())
+            return Digest("jerry", 30)
+
+        def fake_read_config(digest):
+            # fake_read_config shouldn't get called if no config file provided
+            assert coverage_config is not None
+            return DigestContents(
+                [FileContent(path="/dev/null/prelude", content=coverage_config.encode())]
+            )
+
+        mock_gets = [
+            MockGet(
+                product_type=Snapshot,
+                subject_type=PathGlobs,
+                mock=lambda _: Snapshot(Digest("bosco", 30), ("/dev/null/someconfig",), ()),
+            ),
+            MockGet(product_type=DigestContents, subject_type=Digest, mock=fake_read_config,),
+            MockGet(product_type=Digest, subject_type=CreateDigest, mock=fake_handle_config),
+        ]
+
+        result = run_rule(create_coverage_config, rule_args=[coverage], mock_gets=mock_gets)
+        assert result.digest.fingerprint == "jerry"
+        assert len(resolved_config) == 1
+        return resolved_config[0]
+
+    def test_default_no_config(self) -> None:
+        resolved_config = self.run_create_coverage_config_rule(coverage_config=None)
+        assert (
+            resolved_config
+            == "[run]\nbranch = True\nrelative_files = True\nomit = test_runner.pex/*\n\n"
+        )
+
+    def test_simple_config(self) -> None:
+        config = dedent(
+            """
+          [run]
+          branch = True
+          relative_files = True
+          jerry = HELLO
+          """
+        )
+        resolved_config = self.run_create_coverage_config_rule(coverage_config=config)
+        assert (
+            resolved_config
+            == "[run]\nbranch = True\nrelative_files = True\njerry = HELLO\nomit = test_runner.pex/*\n\n"
```

Comment:
Nit: these tests would be more readable if you use `dedent()` for the expected result.

Reply:
I tried it, it didn't work..
<img width="1381" alt="Screen Shot 2020-07-08 at 6 08 16 PM" src="https://user-images.githubusercontent.com/1268088/86985397-0260ef00-c146-11ea-8750-6f5764f53613.png">

Reply:
It should be:
```
dedent(
    """\
    [run]
    branch = True
    """
)

Reply:
that doesn't work, dedent doesn't remove the spaces for some reasons. see:
<img width="1399" alt="Screenshot 2020-07-08 18 54 49" src="https://user-images.githubusercontent.com/1268088/86988208-933ac900-c14c-11ea-9efc-1cf61df0a6be.png">

Reply:
That's because the `[` is not aligned with the below lines like the `b` directly underneath. They need to be perfectly aligned.

Reply:
updated